### PR TITLE
Moved map-based commands to OpossumBot

### PR DESCRIPTION
**Pull request by RossSmyth**

### CODE REVIEW

**Comment by RossSmyth** [comment]
asdf

File: src/PossumLib.hs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+startHandler = do
+    liftIO $ putStrLn "Started"
+
+endHandler :: IO ()
+endHandler = putStrLn "Ended"
+
+eventHandler :: Map.Map Text (Message -> DiscordHandler ())  -> (Message -> Bool) -> Event -> DiscordHandler ()
+eventHandler cmds preCheck event = case event of
+    MessageCreate m -> when (preCheck m) $ do
+        case Map.lookup (Text.drop 1 (messageContent m)) cmds of
```
Comment (RossSmyth) [inline src/PossumLib.hs:50]: Currently this unconditionally drops the first character under the assumption that the prefix is a single character (i.e. `"!"`). This should probably be changed.

**Comment by RossSmyth** [comment]
Since your MySQL driver isn't really ready for primetime I didn't include that logic, but it shouldn't be too hard to add with [this package](https://hackage.haskell.org/package/mysql-haskell-0.8.4.3).

**Comment by RossSmyth** [comment]
Fixed the merge conflicts. LGTM

**Comment by Contrastellar** [comment]
ezgif com-gif-maker

**Pull request closed without merging**